cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -539,14 +540,33 @@ private String pg(CQLTermNode node) throws QueryValidationException {
     if ("cql.allRecords".equalsIgnoreCase(node.getIndex())) {
       return "true";
     }
-    //determine if this table is real by checking in schema
-    //determine right here whether this node deals with a foreign term on right or left side
-    Table indexTable = checkForForeignLocationOfIndex(node);
-    if (indexTable != null) {
-      //we are doing a foreign key search
-      //determine foreign key linking tables
-      return subQuery(node.getIndex(), node, indexTable);
+
+//  //determine if this table is real by checking in schema
+//  //determine right here whether this node deals with a foreign term on right or left side
+//  Table indexTable = checkForForeignLocationOfIndex(node);
+//  if (indexTable != null) {
+//    //we are doing a foreign key search
+//    //determine foreign key linking tables
+//    return subQuery(node.getIndex(), node, indexTable);
+//  }
+
+    // determine if index is in a foreign table
+    if (dbTable != null) {
+      String srcTabName = dbTable.getTableName();
+      String targetTabAlias = node.getIndex().split("\\.")[0];
+      List<DbFkInfo> fks = DbSchemaUtils.findForeignKeysFromSourceTableToTargetAlias(dbSchema, srcTabName, targetTabAlias);
```

Comment:
Can you rename `fks` to a self-explanatory name, for example `foreignKeys`?

Reply:
leaving as is.